### DISCUSSION TITLE
after_create_account_view

### DISCUSSION THREAD

**Discussion by igor-alexandrov** [jeremyevans/rodauth#360] [General]
Hello, I have a scenario in which I don't want to auto login user after creating an account. Instead of this I want to render a page with clarification about what to do next: "Congrats, we've sent you an email with verification link. Blah Blah Blah". I can use `create_account_redirect` for this and create a route that will render this template. But in this case this route and this template will be public available if you know the URL.

I propose implementing  `after_create_account_view` that will be rendered if defined, if it is not defined, than current logic with redirect will work. Seems like this can be easily done with an if-case here: https://github.com/jeremyevans/rodauth/blob/master/lib/rodauth/features/create_account.rb#L83.

Please tell me what do you think. Thanks!

**Comment by janko** [comment]
I needed to do a similar thing when working on a screencast series for rebuilding GitHub authentication with Rodauth. It was about displaying a "password sent" page after password reset request, which I implemented by using `return_response` to render an HTML template:

```rb
after_reset_password_request do
  # renders a "password_reset_sent" view template with a "Password sent!" page title
  return_response view("password_reset_sent", "Password sent!")
end
```

I believe you can do the exact same thing for account creation, by using the `after_create_account` hook to render a template. In that case you don't need to define a route for it, so it won't be publicly accessible.

The `return_response` method will internally halt request handling, so any code below won't get executed. And Sequel will commit the transaction by default when you `throw` inside of it (it's also possible to tell it to rollback by calling `DB.rollback_on_exit` before throwing).

**Reply by igor-alexandrov** [reply]
Yes, I forgot about `return_response`. We should somewhere store all this tips and tricks.

**Reply by janko** [reply]
The [guides](https://rodauth.jeremyevans.net/documentation.html#guides) section is intended to be a place for these how-tos 🙂

**Reply by janko** [reply]
Opened a pull request that adds the guide – https://github.com/jeremyevans/rodauth/pull/361.
